packages/core/src/codewhispererChat/controllers/chat/controller.ts

```diff
@@ -96,10 +96,10 @@ import { amazonQTabSuffix } from '../../../shared/constants'
 import { OutputKind } from '../../tools/toolShared'
 import { ToolUtils, Tool, ToolType } from '../../tools/toolUtils'
 import { ChatStream } from '../../tools/chatStream'
-import { ChatHistoryStorage } from '../../storages/chatHistoryStorage'
 import { tempDirPath } from '../../../shared/filesystemUtilities'
 import { Database } from '../../../shared/db/chatDb/chatDb'
 import { TabBarController } from './tabBarController'
+import { messageToChatMessage } from '../../../shared/db/chatDb/util'
```

Comment:
Need to update this
https://github.com/aws/aws-toolkit-vscode/blob/96f76c38c3fbb26f9b90bc9ed4d741143beac293/packages/core/src/shared/db/chatDb/util.ts#L79

Don't need the whole messageContext

Reply:
Fixed

Reply:
where?

Reply:
Added a `formatChatHistoryMessage` which gets called when adding the message. Like the approach used in chatHistory: https://github.com/aws/aws-toolkit-vscode/blob/0398a10a81aea275b4abb1984112f54ac57d065f/packages/core/src/shared/db/chatDb/chatDb.ts#L336

Reply:
does assistantResponse message also gets added via this?

Reply:
Yeah the user message and assistantResponse both get added to the db using the same `addMessage` method. So that's why I added the check in `formatChatHistoryMessage` for prompt type